### ISSUE TITLE
display hangs but the player still reports active and playing

### ISSUE THREAD

**Issue by tfreedy** [popcornmix/omxplayer#752]
I'm using a Pi4 with Raspbian to stream RTSP from a IP camera. After a couple minutes
the display will freeze but I'm still able to query the player via DBUS.

omxplayer - Commandline multimedia player for the Raspberry Pi
        Build date: Thu, 01 Aug 2019 12:50:43 +0000
        Version   : f543a0d [master]

Test program attached
[rtsp.txt](https://github.com/popcornmix/omxplayer/files/3893748/rtsp.txt)

vcgencmd set_logging level=0xc0
vcdbg log msg

 002903.125: arasan: arasan_emmc_open
002905.300: arasan: arasan_emmc_set_clock C0: 0x00000000 C1: 0x000ef447 emmc: 200000000 actual: 200000 div: 0x000001f4 target: 200000 min: 100000 max: 400000 delay: 10
003022.044: arasan: arasan_emmc_set_clock C0: 0x00000000 C1: 0x000ef447 emmc: 200000000 actual: 200000 div: 0x000001f4 target: 200000 min: 100000 max: 400000 delay: 10
003029.200: arasan: arasan_emmc_set_clock C0: 0x00000f00 C1: 0x000e01c7 emmc: 200000000 actual: 130039 div: 0x00000301 target: 130000 min: 100000 max: 400000 delay: 15
003071.381: arasan: arasan_emmc_set_clock C0: 0x00000f06 C1: 0x000e0307 emmc: 200000000 actual: 33333333 div: 0x00000003 target: 40000000 min: 0 max: 40000000 delay: 1
003368.382: brfs: File read: /mfs/sd/config.txt
003377.020: brfs: File read: 1831 bytes
003430.684: brfs: File read: /mfs/sd/config.txt
003878.686: gpioman: gpioman_get_pin_num: pin DISPLAY_DSI_PORT not defined
003881.176: *** Restart logging
003881.216: brfs: File read: 1831 bytes
003885.047: hdmi: HDMI:hdmi_get_state is deprecated, use hdmi_get_display_state instead
003920.972: HDMI0: hdmi_pixel_encoding: 300000000

003920.984: HDMI1: hdmi_pixel_encoding: 300000000

003931.446: brfs: File read: /mfs/sd/bcm2711-rpi-4-b.dtb
003931.481: Loading 'bcm2711-rpi-4-b.dtb' to 0x100 size 0x9ed3
004154.730: brfs: File read: 40659 bytes
004157.024: brfs: File read: /mfs/sd/config.txt
004157.735: dtparam: audio=on
004191.092: brfs: File read: 1831 bytes
004213.643: brfs: File read: /mfs/sd/overlays/vc4-fkms-v3d.dtbo
004265.126: Loaded overlay 'vc4-fkms-v3d'
004383.127: brfs: File read: 1255 bytes
004386.345: brfs: File read: /mfs/sd/cmdline.txt
004386.425: Read command line from file 'cmdline.txt':
004386.475: 'console=serial0,115200 console=tty1 root=PARTUUID=6c586e13-02 rootfstype=ext4 elevator=deadline fsck.repair=yes rootwait'
005928.326: brfs: File read: 121 bytes
006686.082: brfs: File read: /mfs/sd/kernel7l.img
006686.140: Loading 'kernel7l.img' to 0x8000 size 0x57c1f0
006686.208: Device tree loaded to 0x2eff5b00 (size 0xa428)
006691.401: gpioman: gpioman_get_pin_num: pin SDCARD_CONTROL_POWER not defined
009059.696: vchiq_core: vchiq_init_state: slot_zero = 0xded80000, is_master = 1
009062.564: hdmi: HDMI:hdmi_get_state is deprecated, use hdmi_get_display_state instead
009068.331: TV service:host side not connected, dropping notification 0x00000002, 0x00000001, 0x0000005f
012487.346: brfs: File read: 5751280 bytes
012771.877: TV service:host side not connected, dropping notification 0x00000008, 0x00000004, 0x00000001
065769.943: alloc_compact_internal(space=3168448, in_low_region=0, mode=7) failed
169482.156: clockRIL:waiting for 200000 (200 ms)
169520.145: clock:22:GetConfig(9000004)
169621.956: clock:22:GetConfig(9000004)
169682.188: clockRIL:waiting for 200000 (200 ms)
169723.763: clock:22:GetConfig(9000004)
169825.570: clock:22:GetConfig(9000004)
169882.218: clockRIL:waiting for 200000 (200 ms)
169927.413: clock:22:GetConfig(9000004)
170029.255: clock:22:GetConfig(9000004)
170082.248: clockRIL:waiting for 200000 (200 ms)
170131.068: clock:22:GetConfig(9000004)
170232.882: clock:22:GetConfig(9000004)
170282.277: clockRIL:waiting for 200000 (200 ms)
170334.696: clock:22:GetConfig(9000004)
170436.506: clock:22:GetConfig(9000004)

**Comment by popcornmix** [comment]
omxplayer is being deprecated in favour of VLC. How does VLC behave?
https://www.raspberrypi.org/forums/viewtopic.php?f=29&t=257395

**Comment by tfreedy** [comment]
I'm looking at trying that. does VLC work the same from the CLI without X11?

**Comment by popcornmix** [comment]
Yes, VLC supports fullscreen and windowed playback, and can be run from X or from command line without X.

**Comment by premultiply** [comment]
Such a large VLC footprint for hw decoding on RPi... :-(

**Comment by popcornmix** [comment]
omxplayer was designed to be a very simple player.
But every week there are requests for more and more features and it is no longer simple and still doesn't support all the features requested.
You can switch to VLC and get many new features. Or you can continue using omxplayer with the features it has now.

**Comment by vocqan-2gIbdy-pipguf** [comment]
**aşk**
